laws/src/main/scala/cats/laws/ApplicativeErrorLaws.scala

```diff
@@ -39,6 +39,12 @@ trait ApplicativeErrorLaws[F[_], E] extends ApplicativeLaws[F] {
 
   def attemptFromEitherConsistentWithPure[A](eab: Either[E, A]): IsEq[F[Either[E, A]]] =
     F.attempt(F.fromEither(eab)) <-> F.pure(eab)
+
+  def onErrorPure[A, B](a: A, f: E => F[B]): IsEq[F[A]] =
+    F.onError(F.pure(a)){case err => F.void(f(err))} <-> F.pure(a)
```

Comment:
I've mentioned this in my comments, I couldn't get a `Gen[F[Unit]]`. However, it was my first time dealing with all the discipline infrastructure, and never had to to this particular thing in my own experience with scalacheck, so glad to be proven wrong :)